UnityProject/Assets/Scripts/Input System/CablePlacementVisualisation.cs

```diff
@@ -0,0 +1,239 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// [Client Side Only] MonoBehaviour for handling cable placement visualisation
+/// </summary>
+[RequireComponent(typeof(LineRenderer))]
+public class CablePlacementVisualisation : MonoBehaviour
+{
+	private const int gridSize = 3;
+	private const int maxIndex = gridSize - 1;
+
+	/// <summary>
+	/// color of startPoint(point on which you press mouse button down)
+	/// </summary>
+	[SerializeField] private Color startPointColor;
+	/// <summary>
+	/// color of currentlyHoveredPoint(point below mouse)
+	/// </summary>
+	[SerializeField] private Color onHoverPointColor;
+
+	/// <summary>
+	/// point on which player has pressed mouse button
+	/// </summary>
+	private Vector2Int startPoint;
+	/// <summary>
+	/// point on which player has released mouse button
+	/// </summary>
+	private Vector2Int endPoint;
+	/// <summary>
+	/// point that is currently below mouse
+	/// </summary>
+	private Vector2Int currentlyHoveredPoint;
+
+	/// <summary>
+	/// grid of points
+	/// </summary>
+	private SpriteRenderer[,] connectionPoints = new SpriteRenderer[gridSize, gridSize];
+	/// <summary>
+	/// lineRenderer used to render line between points
+	/// </summary>
+	private LineRenderer lineRenderer;
+
+	/// <summary>
+	/// variable used to restore default color after hover
+	/// </summary>
+	private Color defaultPointColor;
+
+	private void Awake()
+	{
+		// initialize grid
+		for (int y = 0; y < gridSize; y++)
```

Comment:
Logically I think it makes sense if the grid you construct has 0,0 as the centre point rather than the top left, so it's relative to the start point. Alternatively it could be structured to match the Connection enum, to tidy up the logic below in GetConnectionByPoint.

Reply:
I changed the grid type from 2d array to dictionary, but GetConnectionByPoint is still needed to determine which point is under the cursor (9ef7753f67ad74949d5bfa8329a959a114742c88)